<launch>
  <arg name="gui" default="true"/>

  <include file="$(find jsk_perception)/sample/include/play_rosbag_people.xml"/>

  <node name="static_tf_publisher"
        pkg="tf" type="static_transform_publisher"
        args="0.0 0.0 2.0 0.0, 0.785, 1.571 /camera_rgb_optical_frame /dummy_frame 10"/>

  <node name="virtual_camera_mono"
        pkg="jsk_perception" type="virtual_camera_mono">
    <remap from="image" to="/camera/rgb/image_rect_color"/>
    <rosparam>
      frame_id: dummy_frame
      child_frame_id: virtual_camera_frame
      initial_pos: [2.0, 0.0, 0.0]
      initial_rot: [0.5, 0.5, -0.5, -0.5]
    </rosparam>
  </node>

  <group if="$(arg gui)">
    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_perception)/sample/config/sample_virtual_camera_mono.rviz"/>
  </group>
</launch>
